overlord/exportstate/manifest.go

```diff
@@ -0,0 +1,209 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2020-2022 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package exportstate
+
+import (
+	"os"
+	"path/filepath"
+
+	"github.com/snapcore/snapd/dirs"
+	"github.com/snapcore/snapd/release"
+	"github.com/snapcore/snapd/snap"
+)
+
+// Manifest describes content content exported to snaps or the host.
+type Manifest struct {
+	SnapInstanceName          string        `json:"snap-instance-name"`
+	SnapRevision              snap.Revision `json:"snap-revision"`
+	ExportedForSnapdAsVersion string        `json:"exported-for-snapd-as-version,omitempty"`
+
+	// SourceIsHost is only true if provider of the manifest is not a snap but the classic system.
+	// All SourcePath fields, as visible through Sets[*].Exports[*].SourcePath, are absolute names
+	// in the host file system.
+	SourceIsHost bool `json:"source-is-host,omitempty"`
+
+	// Sets describe groups of exported files.
+	Sets map[string]ExportSet `json:"sets"`
```

Comment:
In this map (as well in the `Exports` field of `ExportSet` below) it's not clear what we use as keys. Do we use the `Name` field of the value?

Reply:
Ansering myself: yes, it looks like we do :-)